Fails to build for me on Linux (Ubuntu 18.04, GCC 10):
```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
OPENGL_INC
```
Anything special I have to do for OpenGL support?

With ~opengl, it fails even earlier with:
```
CMake Error at .../opt/spack/linux-ubuntu18.04-x86_64/gcc-10.2.0/cmake-3.19.2-gm7i2pzrgdihoposa7lbrfj4kxzphu63/share/cmake-3.19/Modules/FindPackageHandleStandardArgs.cmake:218 (message):
  Could NOT find OpenGL (missing: OPENGL_opengl_LIBRARY OPENGL_glx_LIBRARY
  OPENGL_INCLUDE_DIR)
```

--- a/var/spack/repos/builtin/packages/3dtk/package.py
+++ b/var/spack/repos/builtin/packages/3dtk/package.py
@@ -34,12 +34,14 @@ class _3dtk(CMakePackage):
 
     generator = 'Ninja'
 
-    depends_on('cmake@3.5:', type='build')
+    depends_on('cmake@3.5:', when='@trunk', type='build')
+    depends_on('cmake@2.6.1:2.999', when='@1.2', type='build')
     depends_on('ninja', type='build')
     depends_on('boost+serialization+graph+regex+filesystem+system+thread+date_time+program_options')
     depends_on('suite-sparse')
     depends_on('zlib')
     depends_on('libpng')
+    depends_on('eigen')
     depends_on('cgal', when='+cgal')
     depends_on('gl', when='+opengl')
     depends_on('glew', when='+opengl')
@@ -66,7 +68,6 @@ class _3dtk(CMakePackage):
             self.define('WITH_GLFW', False),
             self.define('WITH_FTGL', False),
             self.define('WITH_XMLRPC', False),
-            self.define('WITH_EIGEN3', False),
             self.define('WITH_LIBCONFIG', False),
             self.define('WITH_ROS', False),
             self.define('WITH_PYTHON', False),

--- a/var/spack/repos/builtin/packages/3dtk/package.py
+++ b/var/spack/repos/builtin/packages/3dtk/package.py
@@ -19,7 +19,7 @@ class _3dtk(CMakePackage):
     # Repo seems to be in the process of switching to git:
     # https://github.com/3DTK/3DTK
 
-    version('trunk', svn='https://svn.code.sf.net/p/slam6d/code/trunk')
+    version('trunk', svn='https://svn.code.sf.net/p/slam6d/code/trunk', preferred=True)
     version('1.2', svn='https://svn.code.sf.net/p/slam6d/code/branches/3dtk-release-1.2')
 
     variant('cgal', default=False, description='Compile with CGAL support')

--- a/var/spack/repos/builtin/packages/3dtk/package.py
+++ b/var/spack/repos/builtin/packages/3dtk/package.py
@@ -44,7 +44,7 @@ class _3dtk(CMakePackage):
     depends_on('gl', when='+opengl')
     depends_on('glew', when='+opengl')
     depends_on('freeglut', when='+opengl')
-    depends_on('opencv+calib3d+core+features2d+highgui+imgproc+ml', when='+opencv')
+    depends_on('opencv+calib3d+core+features2d+highgui+imgcodecs+imgproc+ml+videoio', when='+opencv')
     # Because concretizer is broken
     depends_on('opencv+flann', when='+opencv')
     depends_on('cuda', when='+cuda')

